### DISCUSSION TITLE
Unable to use ParaTest with SingleStore

### DISCUSSION THREAD

**Discussion by larskoole** [paratestphp/paratest#814] [Q&A]
| Q | A
| --- | ---
| ParaTest version | 7.0
| PHPUnit version | 10.0.11
| PHP version | 8.2

#### Summary

ParaTest creates databases suffixed with _1, _2, etc.
SingleStore also uses those suffixes for internal purposes.
This results in errors when trying to create the databases.

Is there a way to suffix the database names with strings instead? _a -> ... -> _z -> _aa -> _ab

#### Current behavior

ParaTest can't create databases suffixed with integers in SingleStore.

#### How to reproduce: command, code and error stack trace

Run an instance of SingleStore in Docker:
https://github.com/singlestore-labs/singlestoredb-dev-image#how-to-run-the-docker-image-on-linux-or-apple-intel

Change the PHPUnit environment to interact with SingleStore as the DB (just like you would with MySql).

Run tests.

PS: SingleStore is MySql compatible.

**Stack trace from SingleStore itself:**
```
ERROR 1842 ER_INVALID_DB_NAME: Database name (redacted_test_1) conflicts with an existing partition database name (partition databases are formatted as database_#, where # is a number)
```

#### Expected behavior

An option or manual hook to change the suffix to something like _a, _b, etc.

**Comment by Slamdunk** [comment]
> ParaTest creates databases suffixed with _1, _2, etc.

Hi, ParaTest doesn't create any database, it only provides an environment variable unique to each process, and that variable is an integer.

If you need it to be a letter, you can convert that integer with a simple trick like `chr(96 + (int) getenv('TEST_TOKEN'))`; according to the asciitable, the example I just provided works as long as you spawn fewer than 24 processes.

**Comment by larskoole** [comment]
Ah my bad. Is it possible to do this in the `--passthru-php` option?

**Comment by Slamdunk** [comment]
You should to it in your bootstrap

**Comment by larskoole** [comment]
Ok thanks for the help!
